### ISSUE TITLE
What is the shape/structure of the y_pred and y_true argument in loss function when multiple outputs are used?

### ISSUE THREAD

**Issue by yushuinanrong** [fchollet/keras#7491]
Hi,

I want to write a custom loss function involving two different outputs from one network. I'm not sure how to do it because I cannot access each prediction separately. So in Keras, what is the structure of y_pred and y_true when you have multiple outputs and targets?

Any idea is greatly appreciated. 
Thanks!

**Comment by micahprice** [comment]
How did you train your network without knowing what y_true looks like?

With multiple outputs, y_true needs to be a list of labels, or a dictionary with keys for the name of the output layer pointing to a numpy array with dims (batch_size, num_classes). something like `{"occupation": array_with_dims(32, 8), "height": array_with_dims(32,1)}` if occupation had 8 categories.

A more complete example for creating and passing in x, y_true as part of a generator ):
```
def next(...): # py2, could use yield instead

      ...

      batch_x = np.zeros((batch_size, size_of_input), dtype=K.floatx())
      occupation_label = np.zeros((batch_size, num_occupation_classes), dtype=K.floatx())
      height_label = np.zeros((batch_size, 1), dtype=K.floatx())

      for i, input in enumerate(batch_of_inputs):
            batch_x[i]=input
            occupation_label[i,idx_of_occupation_of_input] = 1.
            height_label[i] = height_of_input

     return batch_x, [occupation_label, height_label]
     # OR
     labels_dict = {'occupation' : occupation_label, "height" : height_label
     return batch_x, labels_dict

```
I edited the generator source code in flow_from_directory() for my own purposes, but it depends on the use case.

I would assume that the structure of y_pred will mirror the structure that you give the model from y_true, but I don't know.

**Comment by yushuinanrong** [comment]
@micahprice 
Hi,

Thanks for your reply! This is exactly what I need.
I didn't realize I can pass the y_true in a dictionary format. 
I'll try it out later today.

**Comment by micahprice** [comment]
https://keras.io/getting-started/functional-api-guide/#multi-input-and-multi-output-models

**Comment by ParikhKadam** [comment]
I was going through various issues in order to solve a problem. And I saw this.. I need a similar kind of help here:

https://github.com/tensorflow/tensorflow/issues/24519

Can you help please?
